### DISCUSSION TITLE
colorScheme-function in theme-object

### DISCUSSION THREAD

**Discussion by alexander802** [mantinedev/mantine#3575] [Feature requests]
Wouldn't it be nice to have a function in the theme-object that returns the light/dark-value based on the current colorScheme?

```
const colorScheme = (light, dark) => {
	return colorScheme === 'light' ? light : dark;
}
```

With this it would be a lot easier to write styles:

```
const useStyles = createStyles((theme) => ({
    tabs: {
        backgroundColor: theme.fn.colorScheme(theme.colors.gray[0], theme.colors.dark[5])
    }
}));
```

**Comment by rtivital** [comment]
```ts
backgroundColor: theme.fn.colorScheme(theme.colors.gray[0], theme.colors.dark[5])
backgroundColor: theme.colorScheme === 'dark' ? theme.colors.gray[0] : theme.colors.dark[5]
```
How is it better? 
- It is not significantly shorter
- It requires user to guess what color goes in first: light or dark
- It adds additional constriction: there is no way to flip logic, for example `theme.colorScheme === 'light' ? x : y`

**Reply by alexander802** [reply]
I would prefer it like that to be honest. But at the end of the day it is always a matter of taste.

I tried to add the function by myself to theme.fn, but it's not added. So I assume the builtin-functions are not merged with custom-functions like theme.color for example? Adding to theme.other.fn would be an option that works.

Unfortunately I don't know how to access the current colorScheme in a custom function without passing to the function itself (which would be absurd).

```
const theme = {
    other: {
        fn: {
            colorScheme = (light, dark) => {
                // how to access the colorScheme here???
                return colorScheme === 'light' ? light : dark;
            }
        }
    }
};
```
